Source/WebCore/workers/WorkerMessagingProxy.cpp

```diff
@@ -436,4 +438,25 @@ void WorkerMessagingProxy::terminateWorkerGlobalScope()
         m_scriptExecutionContext = nullptr;
 }
 
+void WorkerMessagingProxy::setAppBadge(std::optional<uint64_t> badge)
+{
+    ASSERT(!isMainThread());
+
+    if (!m_scriptExecutionContext)
+        return;
+
+    m_scriptExecutionContext->postTask([badge = WTFMove(badge), this, protectedThis = Ref { *this }] (auto&) {
+        ASSERT(isMainThread());
```

Comment:
If you don't use postTaskToLoader(), then this assertion won't hold for nested workers.